src/Controller/SoapController.php

```diff
@@ -22,68 +23,83 @@ class SoapController extends AbstractController
      *
      * @param Request $request
      * @param DdbMoreInfoService $ddbMoreInfoService
-     * @param LoggerInterface $statsLogger
+     * @param StatsLoggingService $statsLoggingService
+     *    Statistics logging service
      * @param $projectDir
      *
      * @return Response
      */
-    public function ddbSoap(Request $request, DdbMoreInfoService $ddbMoreInfoService, LoggerInterface $statsLogger, $projectDir): Response
+    public function ddbSoap(Request $request, DdbMoreInfoService $ddbMoreInfoService, StatsLoggingService $statsLoggingService, $projectDir): Response
     {
         $dbcWsdlFile = $projectDir.self::DDB_WSDL_FILE;
 
-        return $this->soap($request, $ddbMoreInfoService, $statsLogger, $dbcWsdlFile);
+        return $this->soap($request, $ddbMoreInfoService, $statsLoggingService, $dbcWsdlFile);
     }
 
     /**
      * @Route("/fbs/2.11/", name="fbs_soap")
      *
      * @param Request $request
      * @param DbcMoreInfoService $dbcMoreInfoService
-     * @param LoggerInterface $statsLogger
+     * @param StatsLoggingService $statsLoggingService
+     *    Statistics logging service
      * @param $projectDir
      *
      * @return Response
      */
-    public function fbsSoap(Request $request, DbcMoreInfoService $dbcMoreInfoService, LoggerInterface $statsLogger, $projectDir): Response
+    public function fbsSoap(Request $request, DbcMoreInfoService $dbcMoreInfoService, StatsLoggingService $statsLoggingService, $projectDir): Response
     {
         $dbcWsdlFile = $projectDir.self::DBC_WSDL_FILE;
 
-        return $this->soap($request, $dbcMoreInfoService, $statsLogger, $dbcWsdlFile);
+        return $this->soap($request, $dbcMoreInfoService, $statsLoggingService, $dbcWsdlFile);
     }
 
     /**
      * @Route("/defaultcover/2.11/", name="default_cover_soap")
      *
      * @param Request $request
      * @param DefaultCoverMoreInfoService $defaultCoverMoreInfoService
-     * @param LoggerInterface $statsLogger
+     * @param StatsLoggingService $statsLoggingService
      * @param $projectDir
      *
      * @return Response
      */
-    public function defaultCoverSoap(Request $request, DefaultCoverMoreInfoService $defaultCoverMoreInfoService, LoggerInterface $statsLogger, $projectDir): Response
+    public function defaultCoverSoap(Request $request, DefaultCoverMoreInfoService $defaultCoverMoreInfoService, StatsLoggingService $statsLoggingService, $projectDir): Response
     {
         $dbcWsdlFile = $projectDir.self::DBC_WSDL_FILE;
 
-        return $this->soap($request, $defaultCoverMoreInfoService, $statsLogger, $dbcWsdlFile);
+        return $this->soap($request, $defaultCoverMoreInfoService, $statsLoggingService, $dbcWsdlFile);
     }
 
     /**
      * Return a SOAP response for the given request.
      *
      * @param Request $request
      * @param AbstractMoreInfoService $moreInfoService
-     * @param LoggerInterface $statsLogger
+     * @param StatsLoggingService $statsLoggingService
```

Comment:
The logger service implements the `LoggerInterface` this should be the type that the function takes.

Reply:
But this is autowired. Therefore it needs to be the precise service.